src/IndexManager.php

```diff
@@ -132,12 +133,18 @@ public function search($query, $className, ObjectManager $objectManager, $page =
                 $entityClass = $className;
             }
 
+            if (!array_key_exists($entityClass, $objectsByEntityClass)) {
+                $objectsByEntityClass[$entityClass] = [];
+            }
+
+            $objectsByEntityClass[$entityClass][] = $id;
+        }
+
+        foreach ($objectsByEntityClass as $entityClass => $ids) {
             $repo = $objectManager->getRepository($entityClass);
-            $entity = $repo->findOneBy(['id' => $id]);
+            $entities = $repo->findById($ids);
```

Comment:
With this approach, we don't loose the *order* of the results?

Reply:
You're right, we loose the order. MySQL return results by id ASC by default. We could use the `ORDER BY FIELD` (https://stackoverflow.com/questions/3799935/mysql-select-where-in-given-order) but not supported by doctrine by default. Any idea @nunomaduro ?

Reply:
Yes. Just re-order the objects again some lines after.

Reply:
@nunomaduro done with this commit: https://github.com/algolia/search-bundle/pull/298/commits/77cf8aa57bda935f850133d5a6dbd64e32f0904c